src/App.tsx

```diff
@@ -15,36 +15,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+function getPreparedGoods(goods: string[], sortParameter: SortType): string[] {
+  if (sortParameter) {
+    goods.sort((a, b) => {
+      switch (sortParameter) {
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return goods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortParameter, setSortParameter] = useState(SortType.NONE);
+
+  const goods = getPreparedGoods([...goodsFromServer], sortParameter);
```

Comment:
u can get this array from import, and make a copy in the middle of the function. Do it
```suggestion
  const goods = getPreparedGoods(sortParameter);
```